authors/apps/authentication/serializers.py

```diff
@@ -214,55 +215,129 @@ class SocialAuthSerializer(serializers.Serializer):
-    password = serializers.CharField(max_length=250, required=True)
-    password_confirm = serializers.CharField(max_length=250, required=True)
+    token = serializers.CharField(
+        max_length=250,
+        write_only=True,
+        allow_null=True,
+        allow_blank=True,
+        required=False
+    )
+    password = serializers.CharField(
+        max_length=250,
+        write_only=True,
+        allow_null=True,
+        allow_blank=True,
+        required=False
+    )
+    password_confirm = serializers.CharField(
+        max_length=250,
+        write_only=True,
+        allow_null=True,
+        allow_blank=True,
+        required=False
+    )
 
     def validate(self, data):
+        if not data.get("token"):
+            raise serializers.ValidationError({
+                "token": "There is no token provided"
+            })
         try:
             token_object = Token.objects.get(key=data.get("token"))
         except ObjectDoesNotExist:
             raise serializers.ValidationError({
-                "token": "invalid token"
+                "token": "Invalid token"
             })
         if self.expired_token(token_object):
             raise serializers.ValidationError({
-                "token": "invalid token"
+                "token": "Expired token"
             })
+        self.get_password_field_erros(data)
         try:
             validate_password(data.get("password"))
         except ValidationError as error:
             raise serializers.ValidationError({
                 "password": list(error)
             })
-        if data["password"] != data["password_confirm"]:
+        policy = PasswordPolicy.from_names(
+            uppercase=2,
+            numbers=2,
+            special=2
+        )
+        if policy.test(data.get("password")):
+            stats = PasswordStats(data.get("password"))
+            if stats.letters_uppercase < 2:
+                raise serializers.ValidationError({
+                    "password": "Your password should have a minimum of 2 uppercase letters"
+                })
+            elif stats.numbers < 2:
+                raise serializers.ValidationError({
+                    "password": "Your password should have a minimum of 2  numbers"
+                })
+            elif stats.special_characters < 2:
+                raise serializers.ValidationError({
+                    "password": "Your password should have a minimum of 2 special characters"
+                })
+        return data
+
+    def get_password_field_erros(self, data):
+        """
+        Captures null, blank and empty fields for password
+        and password confirm
+        """
+        if not data.get("password") and not data.get("password_confirm"):
+            raise serializers.ValidationError({
+                "password": "Please provide your password",
+                "password_confirm": "Please provide your confirmation password"
```

Comment:
I think `Please provide your confirmation password` can be made a better message. For instance `Please confirm your password`

Reply:
Let me update that

Reply:
Updated